src/NuGet.Clients/NuGet.PackageManagement.UI/UserAction.cs

```diff
@@ -21,12 +21,28 @@ private UserAction(NuGetProjectActionType action, string packageId, NuGetVersion
             Version = packageVersion;
         }
 
+        private UserAction(NuGetProjectActionType action, string packageId, NuGetVersion packageVersion, VersionRange versionRange)
+        {
+            Action = action;
+
+            if (string.IsNullOrEmpty(packageId))
+            {
+                throw new ArgumentNullException(nameof(packageId));
+            }
+
+            PackageId = packageId;
+            Version = packageVersion;
+            VersionRange = versionRange;
```

Comment:
Can version be replaced with version range here? 

Note that version range has things such as OriginalString which should probably be how we write to the project file.

Reply:
> In cases when we are trying to install a Custom Version, Version is the best match in the range, that I will calculate when you hit the enter key.

Wouldn't the better approach be to let restore calculate the version we install?
Or was this an explicit decision with PM?

Reply:
As far as I know, this version will not be the one installed, this Version is used for other things (one is knowing if it is prerelease, that is not on the VersionRange), restore will calculate a version and resolve to the best one, I have a test with a similar case.

Reply:
Ok, resolving.